src/driver/safe/core.rs

```diff
@@ -638,25 +711,91 @@ impl<T> CudaSlice<T> {
     pub unsafe fn transmute_mut<S>(&mut self, len: usize) -> Option<CudaViewMut<'_, S>> {
         (len * std::mem::size_of::<S>() <= self.num_bytes()).then_some(CudaViewMut {
             ptr: self.cu_device_ptr,
-            root: &mut self.cu_device_ptr,
             len,
             marker: PhantomData,
         })
     }
+
+    /// Splits the [CudaSlice] into two at the given index, returning two [CudaViewMut] for the two halves.
+    ///
+    /// Panics if `mid > self.len`.
+    ///
+    /// This method can be used to create non-overlapping mutable views into a [CudaSlice].
+    /// ```rust
+    /// # use cudarc::driver::safe::{CudaDevice, CudaSlice, CudaViewMut};
+    /// # fn do_something(view: CudaViewMut<u8>, view2: CudaViewMut<u8>) {}
+    /// # let dev = CudaDevice::new(0).unwrap();
+    /// let mut slice = dev.alloc_zeros::<u8>(100).unwrap();
+    /// // split the slice into two non-overlapping, mutable views
+    /// let (mut view1, mut view2) = slice.split_at_mut(50);
+    /// do_something(view1, view2);
+    /// ```
+    pub fn split_at_mut<'a>(&'a mut self, mid: usize) -> (CudaViewMut<'a, T>, CudaViewMut<'a, T>) {
```

Comment:
Looks like clippy says this lifetime can be elided, other than that, this PR looks great!

Reply:
Ah my bad, didn't have clippy as check on this machine, fixed in most recent commit